android/src/main/java/com/clevertap/rnsignedcallandroid/internal/util/ReactContextHandler.kt

```
@@ -0,0 +1,121 @@
+package com.clevertap.rnsignedcallandroid.internal.util
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.Log
+import com.facebook.infer.annotation.Assertions
+import com.facebook.react.ReactApplication
+import com.facebook.react.ReactInstanceEventListener
+import com.facebook.react.ReactInstanceManager
+import com.facebook.react.ReactNativeHost
+import com.facebook.react.bridge.ReactContext
+
+
+object ReactContextHandler {
+
+  private const val TAG = "ReactContextHandler"
+
+  fun execute(context: Context, runnable: (reactContext:ReactContext) -> Unit) {
+    val reactContext = getReactContext(context)
+    if (reactContext != null) {
+      runnable(reactContext)
+      return
+    }
+    val reactHost: Any = getReactHost(context) ?: {
+      Utils.log(message = "Failed to get React Host")
```

@gupta-shrinath-ct Instead of just logging, throw an exception to the caller of 'execute' method to report the emitted event for which this scenario occurs.

---

I'm curious about how to handle it in sendEmit method of EventEmitter.

---

As discussed, to ensure try catch handling at single place.